tests/chat/test_chat.py

```diff
@@ -61,3 +61,111 @@ def test_get_student_course_chat(self, message_with_student_course):
             student_course_id=message_with_student_course.student_course.student_course_id)
 
         assert message_with_student_course in course_messages
+
+
+@pytest.mark.django_db
+class TestChatAPI:
+    def test_get_student_course_chat(self, client, persist_second_user, student_course0,
+                                     persist_message_with_student_course):
+        client.force_login(user=persist_second_user)
+        response = client.get(f'/chat/courses/{student_course0.student_course_id}')
+        response_message_ids = [m['message_id'] for m in response.data]
+
+        assert persist_message_with_student_course.message_id in response_message_ids
+        assert response.status_code == 200
+
+    def test_get_student_course_chat_not_logged_in(self, client, student_course0, persist_message_with_student_course):
+        response = client.get(f'/chat/courses/{student_course0.student_course_id}')
+        assert response.status_code == 401
+
+    def test_get_student_course_chat_not_in_course(self, client, persist_teacher, student_course0,
+                                                   persist_message_with_student_course):
+        client.force_login(user=persist_teacher)
+        response = client.get(f'/chat/courses/{student_course0.student_course_id}')
+        assert response.status_code == 401
+
+    def test_add_message_to_course_chat(self, client, persist_second_user, student_course0):
+        chat_messages = Message.objects.get_student_course_chat(student_course0.student_course_id)
+        len_messages_before_post = len(chat_messages)
+        client.force_login(user=persist_second_user)
+        client.post('/chat/post-message/', {
+            "courseId": student_course0.student_course_id,
+            "message": "Test",
+        })
```

Comment:
Maybe the message can be a fixture so we won't need to repeat this?

Reply:
Done